WSRequestHandler.cpp

```diff
@@ -1343,6 +1345,141 @@ void WSRequestHandler::HandleSetSceneItemCrop(WSRequestHandler* req) {
 }
 
 /**
+ * Gets the scene specific properties of the specified source item.
+ *
+ * @param {String (optional)} `scene-name` the name of the scene that the source item belongs to. Defaults to the current scene.
+ * @param {String} `item` The name of the source.
+ *
+ * @return TO POPULATE
+ *
+ * @api requests
+ * @name GetSceneItemSceneProperties
+ * @category sources
+ * @since 4.1.3?
+ */
+void WSRequestHandler::HandleGetSceneItemSceneProperties(WSRequestHandler* req) {
+    if (!req->hasField("item")) {
+        req->SendErrorResponse("missing request parameters");
+        return;
+    }
+
+    const char* item_name = obs_data_get_string(req->data, "item");
+    if (!str_valid(item_name)) {
+        req->SendErrorResponse("invalid request parameters");
+        return;
+    }
+
+    const char* scene_name = obs_data_get_string(req->data, "scene-name");
+    obs_source_t* scene = Utils::GetSceneFromNameOrCurrent(scene_name);
+    if (!scene) {
+        req->SendErrorResponse("requested scene doesn't exist");
+        return;
+    }
+
+    obs_sceneitem_t* scene_item = Utils::GetSceneItemFromName(scene, item_name);
+    if (scene_item) {
+        obs_data_t* data = obs_data_create();
+
+        // is name even needed here?
+        obs_data_set_string(data, "name", item_name);
+        
+        // is nested ideal?
+        obs_data_t* pos_data = obs_data_create();
```

Comment:
I use nested objects for position, crop, scale, and bounds. Do we prefer a flat implementation?

Reply:
Nested is probably cleaner. There's already another method using/returning nested properties so it's not like it's disrupting some unspoken rule.